src/exo/LoopIR_scheduling.py

```diff
@@ -2625,45 +2625,27 @@ def DoFuseLoop(proc_cursor, f_cursor, s_cursor):
     return _fixup_effects(ir, fwd)
 
 
-class DoFuseIf(Cursor_Rewrite):
-    def __init__(self, proc_cursor, f_cursor, s_cursor):
-        self.if1 = f_cursor._node
-        self.if2 = s_cursor._node
-
-        super().__init__(proc_cursor)
-
-        self.proc = InferEffects(self.proc).result()
-
-    def map_stmts(self, stmts_c):
-        stmts = [s._node for s in stmts_c]
-        for i, s in enumerate(stmts):
-            if s is self.if1:
-                if i + 1 >= len(stmts) or stmts[i + 1] is not self.if2:
-                    raise SchedulingError(
-                        "expected the two if statements to be "
-                        "fused to come one right after the other"
-                    )
-
-                if1, if2 = self.if1, self.if2
-
-                # check if the loop bounds are equivalent
-                Check_ExprEqvInContext(
-                    self.orig_proc._node, if1.cond, [if1], if2.cond, [if2]
-                )
+def DoFuseIf(f_cursor, s_cursor):
+    proc = f_cursor.get_root()
+    if f_cursor.next() != s_cursor:
+        raise SchedulingError(
+            "expected the two if statements to be fused to come one right after the other"
+        )
 
-                cond = if1.cond
-                body1 = if1.body
-                body2 = if2.body
-                orelse1 = if1.orelse
-                orelse2 = if2.orelse
-                ifstmt = LoopIR.If(
-                    cond, body1 + body2, orelse1 + orelse2, None, if1.srcinfo
-                )
+    if1 = f_cursor._node
+    if2 = s_cursor._node
+    Check_ExprEqvInContext(proc, if1.cond, [if1], if2.cond, [if2])
 
-                return stmts[:i] + [ifstmt] + stmts[i + 2 :]
+    cond = if1.cond
+    body1 = if1.body
+    body2 = if2.body
+    orelse1 = if1.orelse
+    orelse2 = if2.orelse
+    ifstmt = LoopIR.If(cond, body1 + body2, orelse1 + orelse2, None, if1.srcinfo)
 
-        # if we reached this point, we didn't find the if statement
-        return super().map_stmts(stmts)
+    ir, fwd = f_cursor._delete()
```

Comment:
There's definitely a missing compose here.

Find a test case that breaks it and update the code here.